python/hail/tests/test_api.py

```diff
@@ -619,6 +619,15 @@ def test_issue_3654(self):
         ht = ht.annotate(x = [1,2])
         self.assertEqual(ht.aggregate(hl.agg.array_sum(ht.x) / [2, 2]), [5.0, 10.0])
 
+    def test_rekey_correct_partition_key(self):
+            ht = hl.utils.range_table(5)
+            ht = ht.add_index('a')
+            ht = ht.key_by('idx', 'a')
+            ht = ht.annotate(b=ht.idx)
+            ht = ht.key_by('idx', 'b')
+            self.assertEqual(ht.aggregate(agg.sum(ht.idx)), 10)
```

Comment:
Did this fail before the fix? The only way I see to trigger the bug is to `key_by` something between the existing partition key and the existing key. You could load a ((locus), alleles) keyed matrix_table and then key the rows table by locus.